spec/generators/rails-template.rb

```diff
@@ -0,0 +1,269 @@
+      end
+    end
+  RUBY
+
+  file "app/models/wizard.rb", <<~RUBY
+    class Wizard < ApplicationRecord
+      validates :name, length: { minimum: 5 }, presence: true
+
+      enum house: {
+        Gryffindor: 0,
+        Hufflepuff: 1,
+        Ravenclaw: 2,
+        Slytherin: 3,
+      }
+
+      has_one :wand
+      has_many :spell_books
+    end
+  RUBY
+
+  file "app/models/concerns/mythical.rb", <<~RUBY
+    require 'active_support/concern'
+    module Mythical
+      extend ActiveSupport::Concern
+
+      class_methods do
+        def mythicals
+          all.to_a # yeah!
+        end
+      end
+    end
+  RUBY
+end
+
+def create_migrations
+  if ENV["RAILS_VERSION"] == "4.2"
+    migration_superclass = 'ActiveRecord::Migration'
+  else
+    migration_superclass = "ActiveRecord::Migration[#{ENV['RAILS_VERSION']}]"
+  end
+
+  file "db/migrate/20190620000001_create_wizards.rb", <<~RUBY
+    class CreateWizards < #{migration_superclass}
+      def change
+        create_table :wizards do |t|
+          t.string :name
+          t.integer :house
+          t.string :parent_email
+          t.text :notes
+    
+          t.timestamps
+        end
+      end
+    end
+  RUBY
+
+  file "db/migrate/20190620000002_create_wands.rb", <<~RUBY
+    class CreateWands < #{migration_superclass}
+      def change
+        create_table :wands do |t|
+          t.references :wizard, unique: true, null: false
+          t.string :wood_type
+          t.integer :core_type
+    
+          t.timestamps
+        end
+      end
+    end
+  RUBY
+
+  file "db/migrate/20190620000003_create_spell_books.rb", <<~RUBY
+    class CreateSpellBooks < #{migration_superclass}
+      def change
+        create_table :spell_books do |t|
+          t.string :name
+          t.references :wizard
+        end
+      end
+    end
+  RUBY
+
+  file "db/migrate/20190620000004_add_more_column_types_to_wands.rb", <<~RUBY
+    class AddMoreColumnTypesToWands < #{migration_superclass}
+      def change
+        add_column :wands, :flexibility,    :float,   null: false, default: 0.5
+        add_column :wands, :hardness,       :decimal, null: false, precision: 10, scale: 10, default: 5
+        add_column :wands, :reflectance,    :decimal, null: false, precision: 10, scale: 0, default: 0.5
+        add_column :wands, :broken,         :boolean, null: false, default: false
+        add_column :wands, :chosen_at_date, :date
+        add_column :wands, :chosen_at_time, :time
+        # JSON column type is only supported on 5.2 or higher
+        unless ['4.2', '5.0', '5.1'].include?(ENV['RAILS_VERSION'])
+          add_column :wands, :spell_history,  :json
+          add_column :wands, :maker_info,     :json,    null: false, default: '{}'
+        end
+      end
+    end
+  RUBY
+end
+
+def add_sorbet_test_files
+  file "typed-override.yaml", <<~YAML
+    true:
+    - ./sorbet_test_cases.rb
+  YAML
+
+  copy_file "./sorbet_test_cases.rb", "sorbet_test_cases.rb"
```

Comment:
I think this is one instance that creating a symlink makes more sense than copying the file. We may want to add test cases to this file & it should be easy to sync without having to rerun the generation script.